spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/SimpleHostRoutingFilter.java

```diff
@@ -105,51 +86,78 @@
 
 	private ProxyRequestHelper helper;
 	private Host hostProperties;
-	private PoolingHttpClientConnectionManager connectionManager;
+	private ApacheHttpClientConnectionManagerFactory connectionManagerFactory;
+	private ApacheHttpClientFactory httpClientFactory;
+	private HttpClientConnectionManager connectionManager;
 	private CloseableHttpClient httpClient;
+	private boolean customHttpClient = false;
 
 	@EventListener
 	public void onPropertyChange(EnvironmentChangeEvent event) {
-		boolean createNewClient = false;
+		if(!customHttpClient) {
+			boolean createNewClient = false;
 
-		for (String key : event.getKeys()) {
-			if (key.startsWith("zuul.host.")) {
-				createNewClient = true;
-				break;
+			for (String key : event.getKeys()) {
+				if (key.startsWith("zuul.host.")) {
+					createNewClient = true;
+					break;
+				}
 			}
-		}
 
-		if (createNewClient) {
-			try {
-				SimpleHostRoutingFilter.this.httpClient.close();
-			}
-			catch (IOException ex) {
-				log.error("error closing client", ex);
+			if (createNewClient) {
+				try {
+					SimpleHostRoutingFilter.this.httpClient.close();
+				} catch (IOException ex) {
+					log.error("error closing client", ex);
+				}
+				SimpleHostRoutingFilter.this.httpClient = newClient();
 			}
-			SimpleHostRoutingFilter.this.httpClient = newClient();
 		}
 	}
 
-	public SimpleHostRoutingFilter(ProxyRequestHelper helper, ZuulProperties properties) {
+	public SimpleHostRoutingFilter(ProxyRequestHelper helper, ZuulProperties properties,
+			ApacheHttpClientConnectionManagerFactory connectionManagerFactory,
+			ApacheHttpClientFactory httpClientFactory) {
 		this.helper = helper;
 		this.hostProperties = properties.getHost();
 		this.sslHostnameValidationEnabled = properties.isSslHostnameValidationEnabled();
 		this.forceOriginalQueryStringEncoding = properties
 				.isForceOriginalQueryStringEncoding();
+		this.connectionManagerFactory = connectionManagerFactory;
+		this.httpClientFactory = httpClientFactory;
+	}
+
+	public SimpleHostRoutingFilter(ProxyRequestHelper helper, ZuulProperties properties,
+								   CloseableHttpClient httpClient) {
+		this.helper = helper;
+		this.hostProperties = properties.getHost();
+		this.sslHostnameValidationEnabled = properties.isSslHostnameValidationEnabled();
+		this.forceOriginalQueryStringEncoding = properties
+				.isForceOriginalQueryStringEncoding();
+		this.httpClient = httpClient;
+		this.customHttpClient = true;
 	}
 
 	@PostConstruct
 	private void initialize() {
-		this.httpClient = newClient();
-		this.connectionManagerTimer.schedule(new TimerTask() {
-			@Override
-			public void run() {
-				if (SimpleHostRoutingFilter.this.connectionManager == null) {
-					return;
+		if(!customHttpClient) {
```

Comment:
Do you think we need this filter creating an http client anymore? Maybe deprecate it for removal later?